service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/transport/Exceptions.scala

```diff
@@ -193,10 +193,10 @@ final class ExceptionMessage(val name: String, val detail: String) extends Seria
 class TransportException(val errorCode: TransportErrorCode, val exceptionMessage: ExceptionMessage, cause: Throwable) extends RuntimeException(exceptionMessage.detail, cause) {
 
   def this(errorCode: TransportErrorCode, message: String) =
-    this(errorCode, new ExceptionMessage(getClass.getSimpleName, message), null)
+    this(errorCode, new ExceptionMessage(classOf[TransportException].getSimpleName, message), null)
 
   def this(errorCode: TransportErrorCode, cause: Throwable) =
-    this(errorCode, new ExceptionMessage(getClass.getSimpleName, cause.getMessage), cause)
+    this(errorCode, new ExceptionMessage(classOf[TransportException].getSimpleName, cause.getMessage), cause)
```

Comment:
This was not working as expected. I'm not sure why but before the change the `name` in `ExceptionMessage` was always `Predef$` instead of the expected `TransportException`.

Reply:
I ran into this, too, and it was also reported in #425.

It turns out to be a quirk of Java/Scala: you can't reference `this` as an object in a `this` constructor delegation. So `this.getClass()` is not legal here, and `getClass` is intended as a shortcut for `this.getClass()`. In Java, it wouldn't compile at all, but in Scala, it "helpfully" imports and calls `Predef.getClass` instead!

I found this out from https://stackoverflow.com/a/19124742/29470

Reply:
Won't this implementation still mean that exceptions created with these constructors will not deserialize correctly? I wonder if we should deprecate these overloads. They aren't used in the framework, so they're just landmines for users to trip over.

Reply:
Yeah. Extending `TransportException` will cause issues and we should stir people away from doing it until we have a bomb proof alternative. What works in all cases thought is creating a new instance of a `TransportException` where the `cause` is the user exception (it requires a custom exception serializer though, see tests in this PR).

Reply:
Ugly bug ! 😅